compiler-core/src/type_/environment.rs

```diff
@@ -401,6 +401,7 @@ impl<'a> Environment<'a> {
                     name: name.clone(),
                     variables: self.local_value_names(),
                     type_with_name_in_scope,
+                    situation: Some(UnknownValueConstructorErrorSituation::UnknownConstructorName),
```

Comment:
This is being added _always_ when trying to get a value constructor from the environment, but this ticket is for only when it is in a pattern. This needs to be moved to the pattern analysis code instead.